### PR TITLE
Add encryption support to efs module

### PR DESCRIPTION
**Pull request by wimnat**
##### SUMMARY
The current module did not support efs encryption.  This change adds support.

It also adds support for not purging tags as the old module always would.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
efs.py

##### ANSIBLE VERSION

```
2.5
```

##### ADDITIONAL INFORMATION
I also fixed the spacing used in the documentation section to be standard 2 spaces rather than 4

### CODE REVIEW

**Comment by gundalow** [comment]
Hi, please consider the diff that you generate, for example as you've moved `state` in argspec it looks like it's changed.

**Comment by wimnat** [comment]
@gundalow i realise this diff is a little bigger than it could be because of style cleanup but nonetheless i think style cleanup tasks such as alphabetising module parameters is important. :)

**Comment by sivel** [comment]
Please remove the style changes from this PR. Any style changes should be done in a separate PR to separate functional and non functional changes. This enables us to better utilize these changes if we were to cherry pick changes.

Once that is done we can review for potential inclusion.

**Comment by wimnat** [comment]
@sivel  @gundalow all style changes have been removed

**Comment by gundalow** [comment]
@wimnat Thanks for the update. I'll let one of the AWS maintainers review this.
FYI `#ansible-aws` exists on Freenode IRC

File: lib/ansible/modules/cloud/amazon/efs.py
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/efs.py:623]: I'd like to pull @willthames and @ryansb into this conversation because I keep noticing purge_tags with different default values. I don't have a preference for whether this should default to true or false but since there's been some backlash over it I think there should be a documented decision so we can get some uniformity. See https://github.com/ansible/ansible/pull/31427#discussion_r147088175.
a) default to False and break backwards compatibility but err on the side of caution
b) default to True and keep compatibility but let users accidentally remove their tags

File: lib/ansible/modules/cloud/amazon/efs.py
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/efs.py]: .message doesn't exist at all when running Python 3. The exception handling guidelines have also added BotoCoreError to exceptions that should be caught (which do not have a .response attribute). https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-1

**Review by s-hertel — APPROVED** [review]

File: lib/ansible/modules/cloud/amazon/efs.py
Comment (ryansb) [reply lib/ansible/modules/cloud/amazon/efs.py:623]: I think defaulting to `false` is the safest option, and would be consistent with the purge_tags defaults in other modules.

**Review by ryansb — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/amazon/efs.py
Comment (s-hertel) [reply lib/ansible/modules/cloud/amazon/efs.py:623]: I think the majority do default to false. There are four that definitely do not - ec2_vpc_subnet, elb_target_group, elb_application_lb, and ec2_group all default purge_tags to true. Will those remain an anomaly or can we change the default?

**Comment by wimnat** [comment]
@ryansb @s-hertel @willthames 

So I was going to update the GUIDELINES in line with Ryan's comment above but then I saw this - https://github.com/ansible/ansible/pull/26871 which seems to indicate we had this conversation before and went the other way.  Should we change our mind?

File: lib/ansible/modules/cloud/amazon/efs.py
Comment (s-hertel) [reply lib/ansible/modules/cloud/amazon/efs.py:623]: Linking another PR depending on this decision: https://github.com/ansible/ansible/pull/28513#discussion_r139060330

**Comment by s-hertel** [comment]
Thanks for linking that @wimnat, I hadn't seen it.

cc/ @mikedlr Since you wrote that. Do you have thoughts on this?

File: lib/ansible/modules/cloud/amazon/efs.py
Comment (ryansb) [reply lib/ansible/modules/cloud/amazon/efs.py:623]: I see - I do prefer having a `true` default, but had it in my head that the default was `false` most places. Since this module doesn't have backwards compatibility requirements, I'm ok with defaulting to `true` as in the guidelines.

**Review by ryansb — APPROVED** [review]

**Pull request merged**
